src/property.ts

```diff
@@ -215,23 +215,21 @@ export class PropertiesManager {
   }
 
   set(propertiesToSet: Properties): void {
-    this.properties = {
-      ...Object.keys(propertiesToSet).map((propertyName) => [
-        propertyName,
-        get(propertyName),
-      ]),
-      ...this.properties,
-    };
-    Object.entries(propertiesToSet).forEach(([propertyName, propertyValue]) => {
+    for (const [propertyName, propertyValue] of Object.entries(
+      propertiesToSet
+    )) {
+      if (this.properties[propertyName as KnownProperty] !== undefined) {
```

Comment:
shouldn't this only be if it DOES equal undefined? I.e. save a property's value if it's the first time we are setting it in the session?

Reply:
yep, force of habit. Whoops!